complex.c

```diff
@@ -2229,28 +2229,158 @@ string_to_c_strict(VALUE self, int raise)
- *   '1.0@0'.to_c             # => (1+0.0i)
+ * Returns a Complex object:
+ * parses the leading substring of +self+
+ * to extract two numeric values that become the coordinates of the complex object.
+ *
+ * The substring is interpreted as containing
+ * either rectangular coordinates (real and imaginary parts)
+ * or polar coordinates (magnitude and angle parts),
+ * depending on an included or implied "separator" character:
+ *
+ * - <tt>'+'</tt>, <tt>'-'</tt>, or no separator: rectangular coordinates.
+ * - <tt>'@'</tt>: polar coordinates.
+ *
+ * <b>In Brief</b>
+ *
+ * In these examples, we use method Complex#rect to display rectangular coordinates,
+ * and method Complex#polar to display polar coordinates.
+ *
+ *   # Rectangular coordinates.
+ *
+ *   # Real-only: no separator; imaginary part is zero.
+ *   '9'.to_c.rect         # => [9, 0]         # Integer.
+ *   '-9'.to_c.rect        # => [-9, 0]        # Integer (negative).
+ *   '2.5'.to_c.rect       # => [2.5, 0]       # Float.
+ *   '1.23e-14'.to_c.rect  # => [1.23e-14, 0]  # Float with exponent.
+ *   '2.5/1'.to_c.rect     # => [(5/2), 0]     # Rational.
+ *
+ *   # Some things are ignored.
+     'foo1'.to_c.rect      # => [0, 0]         # Unparsed entire substring.
+ *   '1foo'.to_c.rect      # => [1, 0]         # Unparsed trailing substring.
+     ' 1 '.to_c.rect       # => [1, 0]         # Leading and trailing whitespace.
+     *
+ *   # Imaginary only: trailing 'i' required; real part is zero.
+ *   '9i'.to_c.rect        # => [0, 9]
+ *   '-9i'.to_c.rect       # => [0, -9]
+ *   '2.5i'.to_c.rect      # => [0, 2.5]
+ *   '1.23e-14i'.to_c.rect # => [0, 1.23e-14]
+ *   '2.5/1i'.to_c.rect    # => [0, (5/2)]
+ *
+ *   # Real and imaginary; '+' or '-' separator; trailing 'i' required.
+ *   '2+3i'.to_c.rect      # => [2, 3]
+ *   '-2-3i'.to_c.rect     # => [-2, -3]
+ *   '2.5+3i'.to_c.rect    # => [2.5, 3]
+ *   '2.5+3/2i'.to_c.rect  # => [2.5, (3/2)]
+ *
+ *   # Polar coordinates; '@' separator; magnitude required.
+ *
+ *   '1.0@0'.to_c.polar             # => [1.0, 0.0]
+ *   '1.0@'.to_c.polar              # => [1.0, 0.0]
+ *   "1.0@#{Math::PI}".to_c.polar   # => [1.0, 3.141592653589793]
+ *   "1.0@#{Math::PI/2}".to_c.polar # => [1.0, 1.5707963267948966]
+ *
+ * <b>Parsed Values</b>
+ *
+ * The parsing may be thought of as searching
+ * for {numeric literals}[rdoc-ref:syntax/literals.rdoc@Numeric+Literals]
+ * embedded in the substring.
+ *
+ * This section shows how the method parses numeric values from leading substrings.
+ * The examples show real-only or imaginary-only parsing;
+ * the parsing is the same for each part.
+ *
+ *   '1foo'.to_c # => (1+0i)      # Ignores trailing unparsed characters.
+ *   ' 1 '.to_c  # => (1+0i)      # Ignores leading and trailing whitespace.
+ *   'x1'.to_c   # => (0+0i)      # Finds no leading numeric.
+ *
+ *   # Integer literal embedded in the substring.
+ *   '1'.to_c       # => (1+0i)            #
```

Comment:
Extra characters at the end

```suggestion
 *   '1'.to_c       # => (1+0i)
```

Reply:
Fixed.